src/cuda_eval.cpp

```diff
@@ -763,22 +763,32 @@ static void jitc_cuda_render_var(uint32_t index, Variable *v) {
             break;
 
         case VarKind::TexLookup:
-            fmt("    .reg.f32 $v_out_<4>;\n", v);
+            fmt("    .reg.$t $v_out_<4>;\n", v, v);
             if (a3)
-                fmt("    tex.3d.v4.f32.f32 {$v_out_0, $v_out_1, $v_out_2, $v_out_3}, [$v, {$v, $v, $v, $v}];\n",
-                    v, v, v, v, a0, a1, a2, a3, a3);
+                fmt("    tex.3d.v4.$t.f32 {$v_out_0, $v_out_1, $v_out_2, $v_out_3}, [$v, {$v, $v, $v, $v}];\n",
```

Comment:
Out of curiosity: can CUDA also perform texture lookups that produce a f16 output?

Reply:
I dad quick look at the CUDA documentation and found:

> 16-bit floating-point components are promoted to 32 bit float during texture fetching before any filtering is performed.

This also aligns with the fact that the PTX instruction fetch `tld4` doesn't support `f16` destinations. So my guess is that internally a lookup will be f32 and then converted to f16? Still, it's nice that the PTX lookup instruction abstracts away those steps.